include/eve/algo/views/map.hpp

```diff
@@ -0,0 +1,385 @@
+    {
+      return self.load_op(eve::read(self.base));
+    }
+
+    EVE_FORCEINLINE friend void tagged_dispatch(eve::tag::write_, map_iterator self, auto v)
+      requires (!std::same_as<StoreOp, nothing_t>)
+    {
+      eve::write(self.base, detail::bind_store_op<I>(self.store_op)(v));
+    }
+
+    template <relaxed_sentinel_for<I> I1>
+    EVE_FORCEINLINE bool operator==(map_iterator<I1, LoadOp, StoreOp> y) const
+    {
+      return base == y.base;
+    }
+
+    template <relaxed_sentinel_for<I> I1>
+    EVE_FORCEINLINE auto operator<=>(map_iterator<I1, LoadOp, StoreOp> y) const
+    {
+      return spaceship_helper(base, y.base);
+    }
+
+    EVE_FORCEINLINE auto& operator+=(std::ptrdiff_t n)
+    {
+      base += n;
+      return *this;
+    }
+
+    EVE_FORCEINLINE friend std::ptrdiff_t operator-(map_iterator const & x, map_iterator const & y)
+    {
+      return x.base - y.base;
+    }
+
+    EVE_FORCEINLINE friend std::ptrdiff_t operator-(map_iterator const & x, unaligned_me const & y)
+      requires (!std::same_as<I, unaligned_t<I>>)
+    {
+      return x.base - y.base;
+    }
+
+    template <typename Traits>
+    EVE_FORCEINLINE
+    friend auto tagged_dispatch(preprocess_range_, Traits tr, map_iterator f, map_iterator l)
+      requires (!iterator<I>)
+    {
+      return preprocess_range(tr, map_convert(as_range(f.base, l.base), f.load_op, f.store_op));
+    }
+
+    template <typename Traits>
+    EVE_FORCEINLINE
+    friend auto tagged_dispatch(preprocess_range_, Traits tr, map_iterator f, unaligned_me l)
+      requires (!iterator<I>) && (!std::same_as<I, unaligned_t<I>>)
+    {
+      return preprocess_range(tr, map_convert(as_range(f.base, l.base), f.load_op, f.store_op));
+    }
+
+    // eve::iterator -------------
+    EVE_FORCEINLINE auto previous_partially_aligned() const
+      requires iterator<I>
+    {
+      return map_convert(base.previous_partially_aligned(), load_op, store_op);
+    }
+
+    EVE_FORCEINLINE auto next_partially_aligned() const
+      requires iterator<I>
+    {
+      return map_convert(base.next_partially_aligned(), load_op, store_op);
+    }
+
+    static auto iterator_cardinal() requires iterator<I>
+    {
+      return I::iterator_cardinal();
+    }
+
+    template <typename _Cardinal>
+    EVE_FORCEINLINE auto cardinal_cast(_Cardinal N) const
+      requires iterator<I>
+    {
+      return map_convert(base.cardinal_cast(N), load_op, store_op);
+    }
+
+    template<relative_conditional_expr C, decorator S>
+      requires iterator<I> && (!std::same_as<LoadOp, nothing_t>)
+    EVE_FORCEINLINE friend auto tagged_dispatch(eve::tag::load_,
+                                                C c,
+                                                S s,
+                                                eve::as<vw_type>,
+                                                map_iterator self)
+    {
+      auto loaded =  self.load_op(eve::load(drop_alternative(c), s,
```

Comment:
Why do we need to drop the alternative then repalce_ignore afterward instead of just loading with an alternative?
Is it due to how we handle some traits over map?

Reply:
Well. Our alternative is provided for the mapped to type.

Let's say:

```
auto r = eve::algo::map(ints, [](auto v) { return eve::zip(v, v + v); });
// ...

eve::load[ignore_first(1).else_(kumi::tuple{0, 0}](it);
```

So, in order to load with alternative from the underlying we need to apply the inverse of load_op to the provided alternative and then apply the load_op again.
We can't quite do that - so just `replace_ignored`